src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/AuditUtility.cs

```diff
@@ -307,6 +307,40 @@ private static (string severityLabel, NuGetLogCode code) GetSeverityLabelAndCode
                 }
             }
             return result;
+
+            IEnumerable<(LibraryIdentity package, bool isDirect)> WalkAcceptedPackages(RestoreTargetGraph graph)
+            {
+                Stack<(GraphNode<RemoteResolveResult> node, bool isDirect)> stack = new();
+                foreach (var projectNode in graph.Graphs)
```

Comment:
Why not use Flattened instead? 
It should already exclude the reject nodes + no computation.

I think you *might* need to differentiate between direct and transitive there somehow, but cheaper than an exhaustive graph walk.